crypto/cms/cms_sd.c

```diff
@@ -507,6 +507,9 @@ CMS_SignerInfo *CMS_add1_signer(CMS_ContentInfo *cms,
                                          pk, NULL) <= 0) {
             goto err;
```

Comment:
My error injection found an issue here:
`EVP_DigestSignInit_ex` may return FAILURE and return with `*pctx != NULL` at the same time!
That is a problem now because this PR relies on `EVP_MD_CTX_FLAG_KEEP_PKEY_CTX` and
therefore always calls `EVP_PKEY_CTX_free(si->pctx);` on cleanup.
I will probably fix this by adding `si->pctx = NULL;` here.